openqabot/pc_helper.py

```diff
@@ -100,6 +102,133 @@ def apply_publiccloud_pint_image(settings):
     return settings
 
 
+def get_pint_image(name_filter, field, state, query, region=None):
+    """
+    Calculate the most recent image in PINT:
+    1. Compose PINT url using query+state,
+    2. get list of images from the url,
+    3. search the one with name matching the filter,
+    4. return the newest one.
+
+    :param str name_filter passed to get_recent_pint_image
+    :param str field key name in dictionary returned by PINT to be used as ID in the return dictionary
+    :param str state active/inactive used as name of the json file in PINT url
+    :param str query url of PINT, has to have '/' at the end
+    :param str region optional filter, mostly needed by AWS
+    :return Return dictionary has keys NAME and ID
+    """
+    settings = {}
+    try:
+        url = f"{query}{state}.json"
+        log.debug("Pint url:%s name_filter:%s region:%s", url, name_filter, region)
+        images = pint_query(url)["images"]
+        image = get_recent_pint_image(images, name_filter, region=region, state=state)
+        if image is None:
+            raise ValueError(
+                f"Cannot find matching image in PINT with name:[{name_filter}] and state:{state}"
+            )
+        if field not in image.keys() or "name" not in image.keys():
+            raise ValueError(
+                f"Cannot find expected keys '{field}' in the selected image dictionary {image}"
+            )
+        settings["ID"] = image[field]
+        settings["NAME"] = image["name"]
+    except BaseException as e:
+        log.warning(f"get_pint_image handling failed: {e}")
+    log.debug("Return settings:%s", settings)
+    return settings
+
+
+def pint_url(pint_base_url, csp_name):
+    """
+    :param str pint_base_url, usually https://susepubliccloudinfo.suse.com/v1
+    :param str csp_name microsoft, amazon, google, ...
+    :return str PINT url for the images API
+    """
+    url = "/".join([pint_base_url] + [csp_name, "images"]) + "/"
+    log.debug("PINT url:%s", url)
+    return url
+
+
+def sles4sap_pint_gce(name_filter, state, pint_base_url):
+    """
+    Query PINT about GCE images and retrieve the latest one
+    """
+    job_settings = {}
+    ret = get_pint_image(
+        name_filter=name_filter,
+        field="project",
+        state=state,
+        query=pint_url(pint_base_url, "google"),
+    )
+    if any(ret):
+        job_settings["SLES4SAP_QESAP_OS_VER"] = f"{ret['ID']}/{ret['NAME']}"
```

Comment:
I can potentially rename this one to `PUBLIC_CLOUD_IMAGE_NAME`. I'd prefer to do it later as having a dedicated name could help me to debug the final result looking at the triggered job settings.

Reply:
I don't understand how this can benefit. I only see yet another unnecessary variable.

Reply:
renaming is one of the suggestion that I get from Anton yesterday